.pre-commit-config.yaml

```diff
@@ -0,0 +1,38 @@
+repos:
+  - repo: https://github.com/pre-commit/pre-commit-hooks
+    rev: v4.2.0
+    hooks:
+      - id: check-executables-have-shebangs
+      - id: check-merge-conflict
+      - id: check-shebang-scripts-are-executable
+      - id: check-toml
+      - id: check-yaml
+      - id: fix-byte-order-marker
+      - id: no-commit-to-branch
+        args: ["-b", "main"]
```

Comment:
question for maintainers: While I think this is a good idea, do y'all never commit and push direction to `main`? It appears that way, and hopefully there's a [GitHub branch protection rule](https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/defining-the-mergeability-of-pull-requests/about-protected-branches#restrict-who-can-push-to-matching-branches) to prevent.

@xsc27 - I think this is a good idea for forked repos, so contributors are guided away from making commits to their own forked `main` and then get into trouble with upstream merges.